python-avd/pyavd/_eos_designs/structured_config/base/utils.py

```diff
@@ -76,3 +83,66 @@ def _build_source_interfaces(
             )
 
         return source_interfaces
+
+    def _get_vrf_and_source_interface(
```

Comment:
Handle `ipv6_mgmt_ip` according to the server ip version.

Reply:
since this function is generic for all protocols, it will very likely use names instead of IPs, so we cannot rely on IP version expect for dns. So skipping this for now. @gmuloc ?